### PR TITLE
[KernelGen][Nvidia] Add binomial operator with Triton kernel

### PR DESCRIPTION
**Pull request by NineAnnAnn**
## Summary
Adds a Triton kernel for `binomial`. Draws samples from a binomial distribution with the number of trials given by `count` and the success probability given by `prob`.

## Testing
- **Accuracy**: 27 test cases pass (edge cases: p=0→0, p=1→count, count=0→0; statistical validation via KS test)
- **Performance**: Mean speedup **16.3x** (float16/float32/bfloat16, sizes 64²–4096² + wide 1024×65536)

## Performance
Test command: `pytest benchmark/test_binomial.py --level core` (NVIDIA H20)

### binomial_out

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|------|--------------------|-------------------|---------||
| float16 | [64, 64] | 0.205632 | 0.062608 | 3.284 |
| float16 | [256, 256] | 0.220896 | 0.061760 | 3.577 |
| float16 | [1024, 1024] | 2.739584 | 0.137920 | 19.864 |
| float16 | [4096, 4096] | 54.471073 | 1.912224 | 28.486 |
| float16 | [1024, 65536] | 220.656250 | 7.568128 | 29.156 |
| float32 | [64, 64] | 0.182800 | 0.062208 | 2.939 |
| float32 | [256, 256] | 0.194208 | 0.061824 | 3.141 |
| float32 | [1024, 1024] | 2.519536 | 0.138816 | 18.150 |
| float32 | [4096, 4096] | 49.780687 | 1.911616 | 26.041 |
| float32 | [1024, 65536] | 201.677567 | 7.559040 | 26.680 |
| bfloat16 | [64, 64] | 0.207472 | 0.061408 | 3.379 |
| bfloat16 | [256, 256] | 0.223232 | 0.061632 | 3.622 |
| bfloat16 | [1024, 1024] | 2.702752 | 0.137888 | 19.601 |
| bfloat16 | [4096, 4096] | 53.407936 | 1.914272 | 27.900 |
| bfloat16 | [1024, 65536] | 215.831070 | 7.572448 | 28.502 |

| Operator | Arithmetic Mean Speedup |
|----------|------------------------|
| binomial_out | **16.288** |

## Multi-backend Testing
| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---|---|
| Nvidia (H20) | PASS (27 cases) | 16.288 | Primary |
| Tianshu | N/A | — | — |
| Muxi | N/A | — | — |
| Ascend | N/A | — | — |
| Hygon | N/A | — | — |

## Files Changed
- `src/flag_gems/ops/binomial.py`: Triton kernel implementation
- `tests/test_binomial.py`: Accuracy test
- `benchmark/test_binomial.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Math, stage: alpha 5.4)

### CODE REVIEW

File: tests/test_binomial.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+@pytest.mark.parametrize("n, p", [(10, 0.5), (20, 0.3), (100, 0.9), (1, 0.5)])
+def test_binomial(shape, dtype, n, p):
+    count = torch.full(
+        size=shape, fill_value=float(n), dtype=dtype, device=flag_gems.device
+    )
+    prob = torch.full(
+        size=shape, fill_value=float(p), dtype=dtype, device=flag_gems.device
+    )
+
```
Comment (Caeruleann) [inline tests/test_binomial.py]:
```suggestion
    ref_out = torch.binomial(count, ref_prob)
```

File: conf/operators.yaml [hunk 2]
```diff
@@ -1176,6 +1176,29 @@ ops:
...
+      Draws samples from a binomial distribution with the number of trials
+      given by `count` and the success probability given by `prob`.
+    for:
+      - binomial
+    labels:
+      - aten
+      - KernelGen
+    kind:
+      - Math
+    stages:
```
Comment (Caeruleann) [inline conf/operators.yaml:1271]:
```suggestion
    stages:
      - alpha: '5.4'
```

File: conf/operators.yaml [hunk 3]
```diff
@@ -1176,6 +1176,29 @@ ops:
...
+    description: |
+      A variant of binomial() that writes the result into the out tensor.
+    for:
+      - binomial.out
+    labels:
+      - aten
+      - KernelGen
+    kind:
+      - Math
+    stages:
```
Comment (Caeruleann) [inline conf/operators.yaml:1283]:
```suggestion
    stages:
      - alpha: '5.4'
```

File: src/flag_gems/ops/__init__.py [hunk 4]
```diff
@@ -1064,6 +1065,8 @@
     "binary_cross_entropy_backward",
     "binary_cross_entropy_with_logits",
     "bincount",
+    "binomial",
```
Comment (Caeruleann) [inline src/flag_gems/ops/__init__.py:1131]: sort by a-z

**Review by Caeruleann — APPROVED** [review]
